avocado/core/test.py

```diff
@@ -1146,12 +1143,51 @@ def _execute_cmd(self):
         except process.CmdError as details:
             self._log_detailed_cmd_info(details.result)
             raise exceptions.TestFail(details)
-        with open(self.logfile) as logfile:
-            for line in logfile:
-                if self.re_avocado_log.match(line):
-                    raise exceptions.TestWarn("Test passed but there were warnings"
-                                              " on stdout during execution. Check "
-                                              "the log for details.")
+
+        warn_regex = settings.get_value('simpletests.status',
+                                        'warn_regex',
+                                        key_type='str',
+                                        default=None)
+
+        warn_location = settings.get_value('simpletests.status',
+                                           'warn_location',
+                                           default='all')
+
+        skip_regex = settings.get_value('simpletests.status',
+                                        'skip_regex',
+                                        key_type='str',
+                                        default=None)
+
+        skip_location = settings.get_value('simpletests.status',
+                                           'skip_location',
+                                           default='all')
+
+        # Keeping compatibility with 'avocado_warn' libexec
+        for regex in [warn_regex, r'^\d\d:\d\d:\d\d WARN \|']:
```

Comment:
You only want to compile the regexps ones. Perhaps we could lazily do that even for all tests (global variable), what do you think?

Reply:
Sure.

Reply:
Btw it was just brainstorming, not a requirement 😉

Reply:
Actually please don't do this. I thought about this deeper and with JobAPI we might want to have different behavior for different SimpleTests in a single job and caching the regexp might be counterproductive.